### PR TITLE
feat(create2): allow to set ifExists = "continue"

### PR DESCRIPTION
**Pull request by mjlescano**
closes https://linear.app/usecannon/issue/CAN-518/add-option-to-create-a-non-strict-create2-deploy-step

### CODE REVIEW

**Review by FuzzB0t — APPROVED** [review]

**Review by dbeal-eth — CHANGES_REQUESTED** [review]
One potential problem with this setup is that it doesn't allow for the simultaneous setting of the create2 contract address and the no strict option. It makes me feel like we need to have a separate arg all together

File: packages/builder/src/schemas.ts [hunk 1]
```diff
@@ -80,20 +80,29 @@ export const deploySchema = z
...
         /**
-         *    Contract deployer address.
-         *    Must match the ethereum address format
+         * Determines whether to deploy the contract using create2
+         */
+        ifExists: z
+          .enum(['continue'])
+          .optional()
+          .describe(
+            'Determines whether when trying to deploy the contract using CREATE2 it should continue if the target address already exists. Disabled by default'
```
Comment (dbeal-eth) [inline packages/builder/src/schemas.ts:101]: 'When deploying a contract with CREATE2, determines the behavior when the target contract is already deployed (ex. due to same bytecode and salt). Set to `continue` to allow the build to continue if the contract is found to have already been deployed. By default, an error is thrown and the action is halted.'

File: packages/builder/src/steps/deploy.ts [hunk 2]
```diff
@@ -278,7 +278,7 @@ const deploySpec = {
       if (config.create2) {
         const arachnidDeployerAddress = await ensureArachnidCreate2Exists(
           runtime,
-          typeof config.create2 === 'string' ? (config.create2 as viem.Address) : ARACHNID_DEFAULT_DEPLOY_ADDR
```
Comment (dbeal-eth) [inline packages/builder/src/steps/deploy.ts:281]: i think the previous condition here was better

**Review by dbeal-eth — APPROVED** [review]
more descriptive comment

**Pull request merged**